dali/operators/reader/video_reader_resize_op.h

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_READER_VIDEO_READER_RESIZE_OP_H_
+#define DALI_OPERATORS_READER_VIDEO_READER_RESIZE_OP_H_
+
+#include <string>
+#include <vector>
+
+#include "dali/operators/reader/loader/video_loader.h"
+#include "dali/operators/reader/reader_op.h"
+#include "dali/operators/reader/video_reader_op.h"
+#include "dali/operators/image/resize/resize.h"
+
+namespace dali {
+
+class VideoReaderResize : public VideoReader, protected ResizeAttr, protected ResizeBase {
+ public:
+  explicit VideoReaderResize(const OpSpec &spec)
+      : VideoReader(spec),
+        ResizeAttr(spec),
+        ResizeBase(spec),
+        per_video_transform_metas_(batch_size_) {
+    DALI_ENFORCE(dtype_ == DALI_UINT8, "Data type must be UINT8.");
+    ResizeAttr::SetBatchSize(count_);
+    ResizeBase::InitializeGPU(count_, spec_.GetArgument<int>("minibatch_size"));
+    resample_params_.resize(count_);
+  }
+
+  inline ~VideoReaderResize() override = default;
+
+ protected:
+  void SetupSharedSampleParams(DeviceWorkspace &ws) override {}
+
+  void SetOutputShape(TensorList<GPUBackend> &output, DeviceWorkspace &ws) override {
+    TensorListShape<> output_shape(batch_size_, sequence_dim);
+    for (int data_idx = 0; data_idx < batch_size_; ++data_idx) {
+      per_video_transform_metas_[data_idx] = GetTransformMeta(
+          spec_, GetSample(data_idx).frame_shape(), &ws, data_idx, ResizeInfoNeeded());
+      TensorShape<> sequence_shape{count_, per_video_transform_metas_[data_idx].rsz_h,
+                                   per_video_transform_metas_[data_idx].rsz_w, channels_};
+      output_shape.set_tensor_shape(data_idx, sequence_shape);
+    }
+    output.Resize(output_shape);
+  }
+
+  void ShareSingleOutput(int data_idx, TensorList<GPUBackend> &batch_output,
```

Comment:
```suggestion
  void SequenceToTensorList(int data_idx, TensorList<GPUBackend> &batch_output,
```
I'm not sure if ShareSingleOutput tells what it does.

Reply:
Done